File: string/maximum-number-of-words-you-can-type/maximum-number-of-words-you-can-type.ts
function canBeTypedWords(text: string, brokenLetters: string): number {
    const broken = new Array<boolean>(26).fill(false);

    for (let c of brokenLetters) {
        broken[c.charCodeAt(0) - 'a'.charCodeAt(0)] = true;
    }

    let res = 0;
    let shouldCount = true;
    for (let c of text) {
        if (c === ' ') {
            res += Number(shouldCount);
            shouldCount = true;
        } else if (broken[c.charCodeAt(0) - 'a'.charCodeAt(0)]) {
            shouldCount = false;
        }
    }

    return res + Number(shouldCount);
};

File: string/maximum-number-of-words-you-can-type/maximum-number-of-words-you-can-type.go
package maximumnumberofwordsyoucantype

func canBeTypedWords(text string, brokenLetters string) int {
	lookup := make(map[byte]bool)

	for i := 0; i < len(brokenLetters); i++ {
		lookup[brokenLetters[i]] = true
	}

	ignore := false
	count := 0
	for i := 0; i < len(text); i++ {
		if text[i] == ' ' {
			if ignore == false {
				count++
			}
			ignore = false
		}
		if lookup[text[i]] {
			ignore = true
		}
	}

	if ignore == false {
		count++
	}

	return count
}
